packages/frontend/src/features/sqlRunner/store/barChartSlice.ts

```diff
@@ -103,19 +103,15 @@ export const barChartConfigSlice = createSlice({
             action: PayloadAction<{ index: number; label: string }>,
         ) => {
             if (!config) return;
-            if (!config.display) {
-                config.display = {
-                    yAxis: [
-                        {
-                            label: action.payload.label,
-                        },
-                    ],
-                };
-            }
-            if (config?.display?.yAxis) {
-                config.display.yAxis[action.payload.index].label =
-                    action.payload.label;
-            }
+
+            config.display = config.display || {};
+            config.display.yAxis = config.display.yAxis || [];
+
+            const { index, label } = action.payload;
+            config.display.yAxis[index] = {
+                ...config.display.yAxis[index],
+                label,
+            };
```

Comment:
why not

config.display.yAxis[action.payload.index].label =
                    action.payload.label;

Reply:
Because the `yaxis` could be undefined 😅 
I've added this: https://github.com/lightdash/lightdash/issues/10869

Reply:
changed here: https://github.com/lightdash/lightdash/pull/10867/commits/52522dc9ed8fa67bf1e9ab764f1250e2b9f2f3c9